chinese/fill.py

```diff
@@ -530,3 +531,84 @@ def bulk_fill_usage():
         showText(failed_msg, copyBtn=True)
     mw.progress.finish()
     showInfo(msg)
+
+
+def bulk_fill_frequency():
+    prompt = PROMPT_TEMPLATE.format(field_names="<i>frequency</i>", extra_info="")
+
+    progress_msg_template = """
+            <b>Processing:</b> %(hanzi)s<br>
+            <b>Chinese notes:</b> %(has_fields)d<br>
+            <b>Cards with Frequency added:</b> %(filled)d<br>
+            <b>Cards with no Frequency added:</b> %(not_filled)d<br>
+            <b>Failed:</b> %(failed)d"""
+
+    target_fields = config.get_fields(["frequency"])
+    hanzi_fields = config.get_fields(["hanzi"])
+
+    if not askUser(prompt):
+        return
+
+    n_processed = 0
+    n_updated = 0
+    n_failed = 0
+    n_notfilled = 0
+    failed_hanzi = []
+
+    note_ids = mw.col.findNotes("deck:current")
+    mw.progress.start(immediate=True, min=0, max=len(note_ids))
+
+    for i, note_id in enumerate(note_ids):
+        note = mw.col.getNote(note_id)
+        copy = dict(note)
+
+        # Ensure note type has hanzi present
+        hanzi = None
+        if has_any_field(copy, hanzi_fields):
+            hanzi = get_hanzi(copy)
+
+        if has_any_field(copy, target_fields) and hanzi:
+            n_processed += 1
+
+            try:
+                if all_fields_empty(copy, target_fields):
+                    filled = fill_frequency(hanzi, copy)
```

Comment:
Previously `fill_frequency` always returned `None`, so the cards being updated were not listed in the progress window as being updated.